### ISSUE TITLE
SEO enhancements / requests

### ISSUE THREAD

**Issue by MiloCasagrande** [foundriesio/docs#137]
We've been asked to look into some SEO enhancements after a few crawls of our websites have been performed and SEO-related issues have been detected.

I tried to look into solving some of those, but I have no idea how to achieve that with Sphinx (or if it's even possible).

The issues we should try to fix are:
 * Missing canonical link in each page
Pages like [this](https://docs.foundries.io/latest/getting-started/create-compose-app/index.html) can also be accessed and indexed without the trailing `index.html` ([here](https://docs.foundries.io/latest/getting-started/create-compose-app/)) and should contain a [canonical link](https://en.wikipedia.org/wiki/Canonical_link_element) (ideally to the one without the `index.html`) or search engines will consider the content as duplicate, and index it twice. I looked into Sphinx docs, but could not find an easy solution without having to hack the Sphinx theme.
 * Missing `publisher`, `author`, `datePublished` (and possibly others) tags in each page
I'm not even sure if we can or want to fix those. These are parts of what Google call [structured data](https://developers.google.com/search/docs/data-types/article) and results in higher page rankings, better search results etc.... I know what needs to be added to the pages, I have no idea how to achieve that with Sphinx.

**Comment by MiloCasagrande** [comment]
@MatthewCroughan I'm assigning this to you since you probably hacked the most the docs and Sphinx, so any inputs/insights is welcome.

**Comment by MatthewCroughan** [comment]
I believe this is the right config option for our theme. Though if we were ever to change the theme, we'd have to re-investigate this option. It's interesting that the theme has control over how the canonical url is presented. A mistake in architecture for Sphinx probably. I will look into this after I have solved the higher priority tasks.

https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html

**Comment by MiloCasagrande** [comment]
> https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-canonical_url

Yeah, that was my finding as well, but as far as I understood that setting only works for the root url (the homepage), and we need it for all pages.

**Comment by MatthewCroughan** [comment]
For as long as we use the RTD theme, I do not think this is possible. 

https://github.com/readthedocs/readthedocs.org/issues/5208

**Comment by MatthewCroughan** [comment]
This should be fixed by a8eabcd

As can be seen by a grep, each page now has a unique canonical url that points to index.html, which I believe is what we want:

```
matthew@thinkpad ~/git/foundries/docs/build/html (improvements-05) $ grep -r '<link rel="canonical"'
reference-manual/factory/fioctl/fioctl.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/fioctl/fioctl.html"/>
reference-manual/factory/fioctl/index.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/fioctl/index.html"/>
reference-manual/factory/factory-sources.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/factory-sources.html"/>
reference-manual/factory/api-access.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/api-access.html"/>
reference-manual/factory/factory-definition.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/factory-definition.html"/>
reference-manual/factory/factory.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/factory/factory.html"/>
reference-manual/linux/linux-update.html:    <link rel="canonical" href="https://docs.foundries.io/reference-manual/linux/linux-update.html"/>
```
